src/cmd/build.js

```diff
@@ -132,22 +134,27 @@ export async function defaultPackageCreator(
     filter: (...args) => fileFilter.wantFile(...args),
   });
 
-  let extensionName: string = manifestData.name;
-
-  let {default_locale} = manifestData;
-  if (default_locale) {
-    default_locale = default_locale.replace(/-/g, '_');
-    const messageFile = path.join(
-      sourceDir, '_locales',
-      default_locale, 'messages.json'
-    );
-    log.debug('Manifest declared default_locale, localizing extension name');
-    extensionName = await getDefaultLocalizedName({
-      messageFile, manifestData,
-    });
+  let packageName;
+  if (filename) {
+    packageName = `${filename}.zip`;
```

Comment:
I was expecting that filename would be the full filename including the extension, at least if the filename we got has already an extension like ".zip" or ".xpi" (and this would also have the nice side-effect of making web-ext build able to create files already using the .xpi extension).

Reply:
I'm also thinking that we should likely validate a bit the filename we got. 

e.g. we may at least check that it is a simple file name and not a relative or absolute path, so that the file will still be stored in the artifacts dir (as the build command currently does).

From a quick look to the nodejs API docs, it seems to me that [nodejs path.parse]() method may be useful to achieve that.

Reply:
Hey @rpl, do you think something like this would be sufficient?
```js
try {
    const valFilename = path.parse(filename);
    if (valFilename.dir !== '') {
      throw new UsageError(
        'Error validating the filename: filename cannot contain any path'
      );
    } else {
      packageName = `${filename}`;
    }
} catch (error) {
    throw new UsageError(
      `Error validating the filename: ${error}`);
}
if (!packageName || packageName === '') {
    // set default filename
    // ...
}
```

Reply:
Should we also limit the user on the number of extension types he/she can give to the filename?

Reply:
> Should we also limit the user on the number of extension types he/she can give to the filename?

You mean to only allow some kind of extensions (like .zip and .xpi) to be used as the extension in the final filename, right?
yeah that sounds reasonable to me.

> Hey @rpl, do you think something like this would be sufficient?

I was thinking that we may first replace the part to interpolate and then do something like:

```
export function getPackageNameFromTemplate(
  filenameTemplate: string,
  manifestData: ExtensionManifest
): string {
  ...
  const packageName = ...;

  // Validate the resulting packageName string, after interpolating the manifest property
  // specified in the template string.
  const parsed = path.parse(packageName);
  if (parsed.dir) {
    throw new UsageError(
      `Invalid filename template "${filenameTemplate}". ` +
      `Filename "${packageName}" should not contain a path`
    );
  }
  if (!['.zip', '.xpi'].includes(parsed.ext)) {
    throw new UsageError(
      `Invalid filename template "${filenameTemplate}". ` +
      `Filename "${packageName}" should have a zip or xpi extension`
    );
  }

  return packageName;
```